### PR TITLE
Added fetch_channel_emotes & fetch_global_emotes methods

### PR DESCRIPTION
**Pull request by ZebcoWeb**
## Pull request summary
Adding **Channel** and **Global** emotes API to fetch emotes.
- Added `ChannelEmote` and `GlobalEmote` model class
- Added `get_channel_emotes` & `get_global_emotes` to TwitchHTTP
- Added fetch_channel_emotes & fetch_global_emotes to Client

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
    - [ ] I have updated the changelog with a quick recap of my changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: twitchio/models.py [hunk 1]
```diff
@@ -173,6 +173,84 @@ def __init__(self, http: "TwitchHTTP", data: dict):
...
+
+    __slots__ =  '_http' ,'id' ,'name' ,'images' ,'tier' ,'type' ,'set_id' ,'format' ,'scale' ,'theme_mode'
+
+    def __init__(self, http: "TwitchHTTP", data: dict):
+        self._http = http
+        self.id: str = data["id"]
+        self.name: str = data["name"]
+        self.images = data["images"]
+        self.tier = data["tier"]
+        self.type = data["emote_type"]
```
Comment (IAmTomahawkx) [inline twitchio/models.py:210]: These need to be typed

File: twitchio/models.py [hunk 2]
```diff
@@ -173,6 +173,84 @@ def __init__(self, http: "TwitchHTTP", data: dict):
...
+        self.set_id: str = data["emote_set_id"]
+        self.format = data["format"]
+        self.scale = data["scale"]
+        self.theme_mode = data["theme_mode"]
```
Comment (IAmTomahawkx) [inline twitchio/models.py:214]: these need to be typed

File: twitchio/models.py [hunk 3]
```diff
@@ -173,6 +173,84 @@ def __init__(self, http: "TwitchHTTP", data: dict):
...
+    __slots__ =  '_http', 'id', 'name', 'images', 'format', 'scale', 'theme_mode', 'template'
+
+    def __init__(self, http: "TwitchHTTP", data: dict):
+        self._http = http
+        self.id: str = data["id"]
+        self.name: str = data["name"]
+        self.images = data["images"]
+        self.format = data["format"]
+        self.scale = data["scale"]
+        self.theme_mode = data["theme_mode"]
```
Comment (IAmTomahawkx) [inline twitchio/models.py:249]: these need to be typed

File: twitchio/http.py [hunk 4]
```diff
@@ -334,6 +334,12 @@ async def get_bits_board(
     async def get_cheermotes(self, broadcaster_id: str):
         return await self.request(Route("GET", "bits/cheermotes", "", query=[("broadcaster_id", broadcaster_id)]))
 
+    async def get_channel_emotes(self, broadcaster_id: str):
+        return await self.request(Route("GET", "chat/emotes", "", query=[("broadcaster_id", broadcaster_id)]))
```
Comment (IAmTomahawkx) [inline twitchio/http.py:338]: as mentioned, broadcaster_id should be optional here, and the query parameter should be removed if it is `None`

**Comment by IAmTomahawkx** [comment]
also please run black on your changes, as your pr currently fails the lint test

File: twitchio/models.py [hunk 5]
```diff
@@ -173,6 +173,84 @@ def __init__(self, http: "TwitchHTTP", data: dict):
...
+        self.images = data["images"]
+        self.tier = data["tier"]
+        self.type = data["emote_type"]
+        self.set_id: str = data["emote_set_id"]
+        self.format = data["format"]
+        self.scale = data["scale"]
+        self.theme_mode = data["theme_mode"]
+
+    def __repr__(self):
+        return f"<ChannelEmote id={self.id} name={self.name} type={self.type}>"
```
Comment (IAmTomahawkx) [inline twitchio/models.py:217]: This could subclass GlobalEmote

**Comment by IAmTomahawkx** [comment]
It's also been pointed out to me that these new models haven't been added to the models.rst documentation

**Comment by ZebcoWeb** [comment]
I'll try to fix these

**Comment by ZebcoWeb** [comment]
> It's also been pointed out to me that these new models haven't been added to the models.rst documentation

How can I add this model to models.rst?

**Comment by chillymosh** [comment]
> > It's also been pointed out to me that these new models haven't been added to the models.rst documentation
> 
> How can I add these models to models.rst?

docs/reference.rst
Please also make sure you have updated the changelog

**Comment by IAmTomahawkx** [comment]
Hello, all we're missing here is the one mentioned issue with the query parameter in http.py
Other than that this is ready to go 👍

**Comment by IAmTomahawkx** [comment]
I've gone ahead and made the fixes 👍 
Thanks for the PR!

**Review by IAmTomahawkx — APPROVED** [review]

**Comment by ZebcoWeb** [comment]
> I've gone ahead and made the fixes 👍 Thanks for the PR!
Good job!

**Pull request merged**